cuda_core/cuda/core/experimental/__init__.py

```diff
@@ -2,29 +2,64 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-from cuda.core.experimental import utils
-from cuda.core.experimental._device import Device
-from cuda.core.experimental._event import Event, EventOptions
-from cuda.core.experimental._graph import (
+try:
+    import cuda.bindings
+except ImportError:
+    raise ImportError("cuda.bindings 12.x or 13.x must be installed") from None
+else:
+    cuda_major, cuda_minor = cuda.bindings.__version__.split(".")[:2]
+    if cuda_major not in ("12", "13"):
+        raise ImportError("cuda.bindings 12.x or 13.x must be installed")
+
+import importlib
+
+subdir = f"cu{cuda_major}"
+try:
+    versioned_mod = importlib.import_module(f".{subdir}", __package__)
+    # Import all symbols from the module
+    globals().update(versioned_mod.__dict__)
+except ImportError:
+    # This is not a wheel build, but a conda or local build, do nothing
+    pass
+else:
+    del versioned_mod
+finally:
+    del cuda.bindings, importlib, subdir, cuda_major, cuda_minor
+
+import sys  # noqa: E402
+import warnings  # noqa: E402
+
+if sys.version_info < (3, 10):
+    warnings.warn(
+        "support for Python 3.9 and below is deprecated and subject to future removal",
+        category=UserWarning,
```

Comment:
I think instead of `UserWarning` it's more appropriate to use either `DeprecationWarning` or `FutureWarning` (I'm not 100% clear on which one is correct for our usage here)

Reply:
My intend was to drop 3.9 support immediately in the next release (either 0.5.0 or 0.4.1), so I made it as noisy as possible. NumPy has a `VisibleDeprecationWarning` that I wish to exist as a built-in warning.. 😛 The default `DeprecationWarning` won't show up in the end user land. WDYT?

Reply:
```suggestion
        category=FutureWarning,
```

Reply:
`VisibleDeprecationWarning` is simply a subclass of `UserWarning`, that's the reason you see it. 

Indeed, `FutureWarning` is the proper built-in warning to use when users are supposed to see a deprecation.
https://docs.python.org/3/library/exceptions.html#FutureWarning